src/Microsoft.DotNet.Tools.Compiler/Program.cs

```diff
@@ -66,6 +68,8 @@ public static int Main(string[] args)
                 var outputValue = output.Value();
                 var intermediateValue = intermediateOutput.Value();
 
+                var preconditions = new Preconditions();
```

Comment:
It would be preferable if we didn't need to pipe this thing around everywhere. Can we do an upfront check on the preconditions? I'd like to see the logic for determining what preconditions are present to be centralized.